ReactiveCocoa/Swift/Signal.swift

```diff
@@ -170,22 +170,28 @@ public func take<T, E>(count: Int)(signal: Signal<T, E>) -> Signal<T, E> {
 	precondition(count >= 0)
 
 	return Signal { observer in
-		var taken = 0
+		if count == 0 {
+			sendCompleted(observer)
```

Comment:
This kind of short-circuiting won't work until we have something like https://github.com/ReactiveCocoa/ReactiveCocoa/pull/1735, because this will be called before `SignalProducer.startWithSignal` has finished setting up, and before any `Signal` observers have been attached.

There's no real workaround right now, unfortunately. This is why [your test is failing](https://github.com/NachoSoto/ReactiveCocoa/commit/829809a4a299b724558059a9e315d1a6724dbded#commitcomment-9863244).